datumaro/plugins/icdar_format/converter.py

```diff
@@ -13,182 +13,136 @@
+                        continue
+                annotation += '\"%s\"' % ann.caption
+                annotation += '\n'
+            if len(annotation):
+                anno_file = osp.join(self._save_dir,
+                    IcdarPath.TASK_DIR[self._task], subset_name, 'gt.txt')
+                os.makedirs(osp.dirname(anno_file), exist_ok=True)
+                with open(anno_file, 'w') as f:
+                    f.write(annotation)
+
+class IcdarTextLocalizationConverter(Converter):
+    DEFAULT_IMAGE_EXT = IcdarPath.IMAGE_EXT
 
-class IcdarTextLocalizationConverter(IcdarConverter):
     def __init__(self, *args, **kwargs):
-        kwargs['tasks'] = IcdarTask.text_localization
         super().__init__(*args, **kwargs)
+        self._task = IcdarTask.text_localization
 
-class IcdarTextSegmentationConverter(IcdarConverter):
+    def apply(self):
+        for subset_name, subset in self._extractor.subsets().items():
+            for item in subset:
+                if item.has_image and self._save_images:
+                    self._save_image(item, osp.join(self._save_dir,
+                        IcdarPath.TASK_DIR[self._task],
+                        subset_name, IcdarPath.IMAGES_DIR,
+                        item.id + IcdarPath.IMAGE_EXT))
+
+                annotation = ''
+                for ann in item.annotations:
+                    if ann.type == AnnotationType.bbox:
+                        annotation += ' '.join(str(p) for p in ann.points)
+                        if ann.attributes and 'text' in ann.attributes:
+                            annotation += ' \"%s\"' % ann.attributes['text']
+                    elif ann.type == AnnotationType.polygon:
+                        annotation += ','.join(str(p) for p in ann.points)
+                        if ann.attributes and 'text' in ann.attributes:
+                            annotation += ',\"%s\"' % ann.attributes['text']
+                    annotation += '\n'
+                if len(annotation):
+                    anno_file = osp.join(self._save_dir, IcdarPath.TASK_DIR[self._task],
+                        subset_name, osp.dirname(item.id), 'gt_' + osp.basename(item.id) + '.txt')
+                    os.makedirs(osp.dirname(anno_file), exist_ok=True)
+                    with open(anno_file, 'w') as f:
+                        f.write(annotation)
+
+class IcdarTextSegmentationConverter(Converter):
+    DEFAULT_IMAGE_EXT = IcdarPath.IMAGE_EXT
     def __init__(self, *args, **kwargs):
-        kwargs['tasks'] = IcdarTask.text_segmentation
         super().__init__(*args, **kwargs)
+        self._task = IcdarTask.text_segmentation
+
+    def apply(self):
+        for subset_name, subset in self._extractor.subsets().items():
+            for item in subset:
+                if item.has_image and self._save_images:
+                    self._save_image(item, osp.join(self._save_dir,
+                        IcdarPath.TASK_DIR[self._task],
```

Comment:
I don't see a point in saving into `IcdarPath.TASK_DIR[self._task]` anymore.

Reply:
Why? When we do one task, our images folder will not be in the tasks folder. This is normal. But, for example, assets for three tasks will have one folder of images.

Reply:
Such test fails:

```
    def test_can_save_dataset_with_cyrillic_and_spaces_in_filename(self):
        expected_dataset = Dataset.from_iterable([
            DatasetItem(id='кириллица с пробелом',
                image=np.ones((8, 8, 3)), annotations=[
                    Bbox(0, 1, 3, 5),
                ]),
        ])

        for task in [
            IcdarTextLocalizationConverter,
            IcdarTextSegmentationConverter,
            IcdarWordRecognitionConverter,
        ]:
            with self.subTest(subformat=task), TestDir() as test_dir:
                self._test_save_and_load(expected_dataset,
                    partial(task.convert, save_images=True),
                    test_dir)
```

Reply:
In the extractor `root = format_root/task`. Maybe also do it in the converter?

Reply:
This is what I see in https://github.com/openvinotoolkit/datumaro/pull/174/files#diff-8dabf26e19c18cd14fd53c6edd8a4b1d2dda9ebff6029ef2786e570df53ac0daR76-R77:
and https://github.com/openvinotoolkit/datumaro/pull/174/files#diff-8dabf26e19c18cd14fd53c6edd8a4b1d2dda9ebff6029ef2786e570df53ac0daR123-R124
```
            image_path = osp.join(self._path, IcdarPath.IMAGES_DIR,
                item_id + IcdarPath.IMAGE_EXT)
```

Reply:
Yes, but the path is modified in `find_sources()`. In the extractor `self._path = format_dir/task` (in localization and segmentation tasks). Maybe in the converter to `__init__` with `self._save_dir` do the same.

It seems to me that the images folder should be in the task folder.